src/SaveObject.ts

```diff
@@ -220,11 +220,13 @@ class BitburnerSaveObject {
     }
 
     if (!decodedSaveData || decodedSaveData === "") {
-      return Promise.reject(new Error("Save game is invalid"));
+      console.error("decodedSaveData:", decodedSaveData);
+      return Promise.reject(new Error("Save game is invalid. The save data cannot be decoded."));
     }
 
     let parsedSaveData;
     try {
+      // Typecasting here is fine. We will validate parsedSaveData immediately after this line.
```

Comment:
I don't think we actually need this cast, or it should be possible to get rid of with some code tweaks. The "if" block *does* do the same thing; TS should be able to see from the control-flow that the type is narrowed to this structure after this point. 

Doing it without the cast ensures we haven't missed an edge case.